### PR TITLE
f/github storage

### PR DESCRIPTION
**Pull request by isaingard**
sync with github now (not azure)

### CODE REVIEW

File: ensemble.py [hunk 1]
```diff
@@ -17,6 +17,7 @@
 	storage_path = os.path.join(os.environ['CFFIXED_USER_HOME'], 'Documents')
 else:
 	storage_path = ''
+url = 'https://api.github.com/repos/crescendosw'
```
Comment (michael-hewitt) [inline ensemble.py]: duplicates URL in storage.py.  The url should *only* be in storage.py.

File: ensemble.py [hunk 2]
```diff
@@ -41,10 +42,11 @@ def will_close(self):
...
 
 def sync_songs_and_tunes():
-	hymnals_path = os.path.join(storage_path, 'hymnals')
-	this_week_path = os.path.join(storage_path, 'this_week')
-	threading.Thread(storage.iPad(hymnals_path, None, 'hymnals').synchronize()).run()
-	threading.Thread(storage.iPad(this_week_path, None, 'this-week').synchronize()).run()
+	l_hymnals_path = os.path.join(storage_path, 'hymnals')
+	l_this_week_path = os.path.join(storage_path, 'this_week')
+	r_hymnals_path = os.path.join(url, 'hymnals/contents/Cantus Christi')
+	threading.Thread(storage.iPad(l_hymnals_path, r_hymnals_path).synchronize()).run()
```
Comment (michael-hewitt) [inline ensemble.py]: Here, we should pass in the repository to synchronize.  In this case 'hymnals'.  storage.py should formulate the full URL based on this repository.  Also, Cantus Christi should not be specified.  'contents' will give back a json file with the three top level directories (Cantus Christi, Common, Hymnals).

File: ensemble.py [hunk 3]
```diff
@@ -41,10 +42,11 @@ def will_close(self):
...
+	#threading.Thread(storage.iPad(l_this_week_path, None).synchronize()).run()
```
Comment (michael-hewitt) [inline ensemble.py]: Remove parenthesis from synchronize to background this thread.  However, we will need to make sure they don't press the button twice and kick off two simultaneous synchronizations.

File: storage.py [hunk 4]
```diff
@@ -79,453 +79,243 @@
 - How to detect that an Internet connection is available?
 """
 
+"""
+Design:
```
Comment (michael-hewitt) [inline storage.py]: WAY TO GO UPDATING THE DESIGN!!! Extra super cudos!

File: storage.py [hunk 5]
```diff
@@ -79,453 +79,243 @@
...
 
-service = azure.storage.file.FileService(
-    account_name='worshiptogether',
-    account_key='33TG6/7zK8TmKHmlSRthHpGxve8YJfu3M9ut77vn0lUy'
-                'B2ZQqfL8ZdDiucbB8MAyg59707Mcxywhy2fFG/ISZA==')
 
 last_upload_filename = 'last_upload'
 last_download_filename = 'last_download'
+contents_filename = 'content.json'
+github_dir = 'test/contents/dir/'
```
Comment (michael-hewitt) [inline storage.py]: This should be test_repo = 'test'.  Synchronize should take a repo name and formulate the URL.  It should not synchronize 'dir', but rather should synchronize the contents of the repository, which will list 'dir' as a directory.

File: storage.py [hunk 6]
```diff
@@ -79,453 +79,243 @@
...
+url = 'https://api.github.com/repos/crescendosw'
+
+r_dir = os.path.join(url, github_dir)
```
Comment (michael-hewitt) [inline storage.py]: Unneeded

File: storage.py [hunk 7]
```diff
@@ -79,453 +79,243 @@
 - How to detect that an Internet connection is available?
```
Comment (michael-hewitt) [inline storage.py]: Be sure to delete old design.

File: storage.py (hunk 6 above)
Comment (isaingard) [reply storage.py]: it is used at the bottom

**Pull request closed without merging**